From b1f3d01f03bae04d7bc704556d7a00f124f24488 Mon Sep 17 00:00:00 2001
From: YunQiang Su <wzssyqa@gmail.com>
Date: Thu, 29 Aug 2013 13:05:53 +0800
Subject: [PATCH] Fix JavaScriptCore ftbfs on mips64(el)

Task-number: QTBUG-33187
(cherry picked from qtscript/fdb622dad4d85894c5979b90ef2918307a8c9a8e)
Change-Id: I2612d17ba159474595335adde8429a50e844dd4b
---
 src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h b/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
index 17c06bf..f98fb72 100644
--- a/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
+++ b/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
@@ -325,7 +325,15 @@
 
 #endif /* ARM */
 
+/* CPU(MIPS64) - MIPS 64-bit both BIG and LITTLE endian */
+#if (_MIPS_SIM == _ABI64)
+	#define WTF_CPU_MIPS64 1
+#endif
 
+/* CPU(MIPSN32) - MIPS N32 ABI both BIG and LITTLE endian */
+#if (_MIPS_SIM == _ABIN32)
+#define WTF_CPU_MIPSN32 1
+#endif
 
 /* ==== OS() - underlying operating system; only to be used for mandated low-level services like 
    virtual memory, not to choose a GUI toolkit ==== */
@@ -873,7 +881,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64)
+#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64) || CPU(MIPS64)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64)
 #define WTF_USE_JSVALUE32 1
-- 
1.8.4.rc3

